src/lib/operators/insert.cpp

```diff
@@ -153,7 +153,9 @@ std::shared_ptr<const Table> Insert::_on_execute(std::shared_ptr<TransactionCont
       // Make sure the MVCC data is written before the first segment (and, thus, the chunk) is resized.
       std::atomic_thread_fence(std::memory_order_seq_cst);
 
-      // Grow data Segments.
+      // "Grow" data segments: Segments are pre-allocated during construction, we resize() to make default-constructed
+      // cells visible so that they can be written in the next Insert step (note, even though those cells are now
+      // visible before being written to, they are still invisible from an MVCC point of view).
```

Comment:
```suggestion
      // "Grow" data segments: Segments are pre-allocated during construction. We resize() to make default-constructed
      // cells visible so that they can be written in the next Insert step. Note that those cells are still invisible 
      // from an MVCC point of view until they are actually being written and committed. 
```